cloudinary-image-management-and-manipulation-in-the-cloud-cdn/js/src/deactivate.js

```diff
@@ -0,0 +1,85 @@
+/* global window wp cloudinaryApi */
+
+const Deactivate = {
+	// The link that triggers the ThickBox
+	tbLink: document.getElementById( 'cld-deactivation-link' ),
+	// The different links to deactivate the plugin.
+	pluginListLinks: document.querySelectorAll( '.cld-deactivate-link, .cld-deactivate' ),
+	// The reasons.
+	options: document.querySelectorAll( '.cloudinary-deactivation input[type="radio"]' ),
+	// The feedback submit button.
+	submitButton: document.querySelector( '.cloudinary-deactivation .button-primary' ),
+	// The skip button.
+	skipButton: document.querySelector( '.cloudinary-deactivation .button-link' ),
+	// Selected reason.
+	reason: '',
+	// The more details .
+	more: null,
+	// The deactivation link for the plugin.
+	deactivationUrl: '',
+
+	addEvents: function() {
+		const context = this;
+
+		// Add event listener to deativation links to add the pop up.
+		[ ...context.pluginListLinks ].forEach( ( link ) => {
+			link.addEventListener( 'click', function( ev ) {
+				ev.preventDefault();
+				context.deactivationUrl = ev.target.getAttribute( 'href' );
+				context.tbLink.click();
+				document.getElementById( 'TB_window' ).setAttribute(
+					'style',
+					'bottom: 0;' +
+					'height: 450px;' +
+					'left: 0;' +
+					'margin: auto;' +
+					'right: 0;' +
+					'top: 0;' +
+					'visibility: visible;' +
+					'width: 550px;'
+				);
+			});
+		});
+
+		// Add event listener to skip feedback.
+		context.skipButton.addEventListener( 'click', function( ev ) {
+			window.location.href = context.deactivationUrl;
+		});
+
+		// Add event listener to update reason and more container.
+		[ ...context.options ].forEach( ( option ) => {
+			option.addEventListener( 'change', function( ev ) {
+				context.submitButton.removeAttribute( 'disabled' );
+				context.reason = ev.target.value;
+				context.more = ev.target.parentNode.querySelector( 'textarea' );
+			});
+		});
+
+		// Add event listener to submit the feedbacl.
+		context.submitButton.addEventListener( 'click', function( ev ) {
+			wp.ajax.send( {
+				url: cloudinaryApi.endpoint,
+				data: {
+					reason: context.reason,
+					more: context.more.value,
```

Comment:
@pereirinha - can't read value of null when an option without textarea selected, which stops submission.

Reply:
@DavidCramer 

Good catch. This was addressed.